app/models/target.rb

```diff
@@ -30,12 +30,39 @@ class Target < ApplicationRecord
 
   validate :limit_number_of_targets
 
+  reverse_geocoded_by :latitude, :longitude
+
+  after_create :check_matching_targets
+
   private
 
   def limit_number_of_targets
     return if user && user.targets.count < 10
 
-    errors.add(:base, I18n
-      .t('api.error.invalid_request.max_target_limit'))
+    errors.add(:base,
+               I18n.t('api.error.invalid_request.max_target_limit'))
+  end
+
+  def check_matching_targets
+    matching_users = []
+    Target.where("user_id != #{user.id}").map do |target|
```

Comment:
You can use something like: `Target.where.not(user_id: user.id)`. Also, I would move this to a scope.
Also, you can add in the where the topic for the target so you don't need to load all targets: `Target.where.not(user_id: user.id).where(topic_id: topic_id)`